core/vm/evm.go

```diff
@@ -75,6 +75,8 @@ type BlockContext struct {
 	Time        *big.Int       // Provides information for TIME
 	Difficulty  *big.Int       // Provides information for DIFFICULTY
 	BaseFee     *big.Int       // Provides information for BASEFEE
+	Random      common.Hash    // Provides information for RANDOM
```

Comment:
Couldn't we have this be `*common.Hash` and use it as a signal for pre/post-merge-ness?